### PR TITLE
Pretty print answers errors, add debug option

### PR DESCRIPTION
**Pull request by Delerme**
When debug is true, ugly print errors

### CODE REVIEW

File: src/answers-umd.js [hunk 1]
```diff
@@ -129,6 +129,20 @@ class Answers {
...
+        console.error(e.error.toString());
+        if (config.debug) {
+          console.log({ ...e.error });
+        }
+      }
+    };
+
+    // Print detailed error messages
+    window.addEventListener('error', printError);
+    window.addEventListener('unhandledrejection', printError);
```
Comment (b1lly) [inline src/answers-umd.js]: Can we move this to a helper? Or better yet, into the ErrorReporter implementation?

If the ErrorReporter is mainly to be a networking an abstraction, i'd suggest creating a separate ErrorReporting implementation that encapsulates this DOM integration and uses the ErrorReportingApi for sending data.

That way lines 132-153 are all encapsulated in the implementation itself.

**Review by b1lly — CHANGES_REQUESTED** [review]

File: src/answers-umd.js (hunk 1 above)
Comment (Delerme) [reply src/answers-umd.js]: done

File: src/answers-umd.js [hunk 2]
```diff
@@ -129,11 +129,16 @@ class Answers {
...
+    // Report errors to console & server
+    this._errorReporter = new ErrorReporter({
+      apiKey: config.apiKey,
+      experienceKey: config.experienceKey,
+      experienceVersion: config.experienceVersion,
+      printVerbose: config.debug,
+      sendToServer: !config.suppressErrorReports
+    });
+    window.addEventListener('error', e => this._errorReporter.report(e.error));
+    window.addEventListener('unhandledrejection', e => this._errorReporter.report(e.error));
```
Comment (b1lly) [inline src/answers-umd.js]: I'd consider these DOM dependencies to be part of the "ErrorReporting" implementation. Thoughts?

**Review by b1lly — CHANGES_REQUESTED** [review]
Mmm... that looks so much better!

File: src/answers-umd.js (hunk 2 above)
Comment (Delerme) [reply src/answers-umd.js]: done

**Review by b1lly — APPROVED** [review]

**Pull request merged**